packages/use-debounced-state/src/useDebouncedState.ts

```diff
@@ -0,0 +1,29 @@
+import * as React from 'react';
+
+export function useDebouncedState<T = unknown>(
+  defaultValue: T,
+  wait: number,
+  options = { leading: false },
+) {
+  const [value, setValue] = React.useState(defaultValue);
+  const timeoutRef = React.useRef<number | null>(null);
+  const leadingRef = React.useRef(true);
+
+  const clearTimeout = () => void window.clearTimeout(timeoutRef.current);
```

Comment:
Of the handful of Typescript/lint problems, I'm unsure how best to resolve this one. @branclar ideas? 
![image](https://user-images.githubusercontent.com/3459902/217386014-b9dc6a83-1c58-4361-b769-30c02f4521e4.png)